test/kubernetes_unittest.cc

```diff
@@ -744,4 +760,241 @@ TEST_F(KubernetesTest, KubernetesNamespace) {
   EXPECT_EQ("the-namespace", KubernetesNamespace(reader));
 }
 
+TEST_F(KubernetesTest, QueryMaster) {
+  testing::FakeServer server;
+  server.SetResponse("/a/b/c", "{\"hello\":\"world\"}");
+
+  Configuration config(std::istringstream(
+    "KubernetesEndpointHost: " + server.GetUrl() + "\n"
+  ));
+  Environment environment(config);
+  KubernetesReader reader(config, nullptr);  // Don't need HealthChecker.
+  EXPECT_EQ(QueryMaster(reader, "/a/b/c")->ToString(), "{\"hello\":\"world\"}");
+
+  EXPECT_TRUE(ThrowsQueryException([&reader]{ QueryMaster(reader, "/d/e/f"); }));
```

Comment:
Why not just `EXPECT_THROW`?

Reply:
The problem is that QueryException is a private nested class. Is there some way to make it accessible from this test?

Reply:
Yep. Add `using QueryException = KubernetesReader::QueryException;` in the `protected` section of the test fixture, and you can use it in subclasses.

Reply:
Done. (Thanks!)